### ISSUE TITLE
How to apply ssl-config to server side in Akka HTTP

### ISSUE THREAD

**Issue by tindzk** [typesafehub/ssl-config#22]
For testing purposes I've set the following options in my `application.conf`:

```
akka {
  ssl-config.ssl.debug.all=true
  ssl-config.ssl.loose.acceptAnyCertificate=true
  ssl-config.ssl.loose.allowWeakProtocols=true
  ssl-config.ssl.loose.acceptAnyCertificate=true
  ssl-config.ssl.default=true
  ssl-config.ssl.debug {
    ssl = true
    sslctx = true
    defaultctx = true
  }
}
```

I instantiate the server as follows:

``` scala
  def server(): Future[ServerBinding] =
    Http().bindAndHandle(route,
      interface = Configuration.serverInterface,
      port = Configuration.serverPort,
      httpsContext = Some(httpsContext),
      settings = ServerSettings(system.settings.config))
```

As the debug output indicates, the above settings are getting ignored:

```
[DEBUG] [12/22/2015 15:38:45.904] [main] [AkkaSSLConfig(akka://...)] Initializing AkkaSSLConfig extension...
[DEBUG] [12/22/2015 15:38:45.914] [main] [AkkaSSLConfig(akka://...)] hostnameVerifier: com.typesafe.sslconfig.ssl.DefaultHostnameVerifier@2cc44ad
```

When I print out `system.settings.config.getConfig("akka.ssl-config")` before the server instantiation, it shows my settings as expected.

**Comment by ktoso** [comment]
FYI, we'll want to remove `acceptAnyCertificate` etc https://github.com/typesafehub/ssl-config/issues/25

Having a look at this now...

**Comment by ktoso** [comment]
Why do you provide an HTTPS context explicitly if you aim to allow `ssl-config` to configure it?

ssl-config configures the default context, here you've passed in one explicitly, so ssl-config is circumvented:

```
  private def effectiveHttpsContext(ctx: Option[HttpsContext]): Option[HttpsContext] =
    ctx orElse Some(defaultClientHttpsContext)
```

**Comment by ktoso** [comment]
We may actually not apply `effectiveHttpsContext ` in all right places... I'll ping here soon.

**Comment by ktoso** [comment]
Well, having that said (and less multi-tasking on my end...), it wasn't really pulled in for 2.0 for the server side, it was pulled in because we needed it for the client side's hostname verification...

There's API questions to be asked how we expose this, since it would be good to be "the default TLS settings for the server". At the same time, it's not quite prime time ready for server mode as @wsargent mentioned elsewhere. On the other hand, it's better than nothing.

I have mixed feelings what to do in the Akka 2.0.1 release with this, as it technically may either add APIs or change semantics. The answer to this ticket however is "that's not what it does", I see how you were mislead into thinking it does though (myself for a second got mislead seeing your question).

**Comment by checat** [comment]
I'm trying Akka HTTP with Play right now, and I've found from reference config that config is read under "ssl-config", instead of "ssl-config.ssl".
Documentation refers to "ssl-config.ssl", but reference.conf does not, see
https://github.com/typesafehub/ssl-config/blob/master/ssl-config-core/src/main/resources/reference.conf

**Comment by ktoso** [comment]
Please ask such questions on the mailing list @checat.

As explained here, SSL-Config is not ready to support the server side properly.
You can make use of it if you want to, but it does not do all the things it should.

See documentation http://doc.akka.io/docs/akka/2.4.8/scala/http/server-side-https-support.html#serversidehttps-scala

**Comment by checat** [comment]
@ktoso There was no question from my side. If you think that my observation is not relevant to this report, I can open a separate bug report: `reference.conf` uses just `ssl-config` prefix, where all documentation refers to `ssl-config.ssl` prefix.

**Comment by ktoso** [comment]
Then it's a docs bug, in ssl-config, not in Akka. Please open a ticket there, thanks!

**Comment by checat** [comment]
Ok, it looks already fixed in e989b67e93e1856b38f037023a3b2ddaee48368a, just not released/deployed to http://typesafehub.github.io/ssl-config/ yet.
Sorry for noise.

**Comment by ktoso** [comment]
This was already released I think, closing.
